tools/c7n_kube/c7n_kube/actions/core.py

```diff
@@ -0,0 +1,136 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+from copy import deepcopy
+
+from c7n.actions import Action as BaseAction
+from c7n.utils import local_session, chunks, type_schema
+from c7n.exceptions import PolicyValidationError
+
+from kubernetes.client import V1DeleteOptions
+
+log = logging.getLogger('custodian.k8s.actions')
+
+
+class Action(BaseAction):
+    pass
+
+
+class MethodAction(Action):
+    method_spec = ()
+    chunk_size = 20
+
+    def validate(self):
+        if not self.method_spec:
+            raise NotImplementedError("subclass must define method_spec")
+        return self
+
+    def process(self, resources):
+        session = local_session(self.manager.session_factory)
+        m = self.manager.get_model()
+        client = session.client(m.group, m.version)
+        for resource_set in chunks(resources, self.chunk_size):
+            self.process_resource_set(client, resource_set)
+
+    def process_resource_set(self, client, resources):
+        op_name = self.method_spec['op']
+        for r in resources:
+            log.info('%s %s' % (op_name, r))
+        pass
+
+
+class PatchAction(MethodAction):
+    """
+    Patches a resource
+
+    Requires patch and namespaced attributes on the resource definition
+    """
+    def validate(self):
+        if not self.manager.get_model().patch:
+            raise PolicyValidationError('patch attribute not defined for resource')
+        return self
+
+    def get_permissions(self):
+        patch = self.manager.get_model().patch
+        return ''.join([a.capitalize() for a in patch.split('_')])
+
+    def patch_resources(self, client, patch_args_template, resources):
+        op = getattr(client, self.manager.get_model().patch)
+        patch_args = deepcopy(patch_args_template)
+        for r in resources:
+            patch_args['name'] = r['metadata']['name']
+            if 'namespace' in patch_args_template:
+                patch_args['namespace'] = r['metadata']['namespace']
+            op(**patch_args)
+
+
+class DeleteAction(MethodAction):
+    """
+    Deletes a resource
+
+    Requires delete and namespaced attributes on the resource definition
+    """
+    def validate(self):
+        if not self.manager.get_model().delete:
+            raise PolicyValidationError('delete attribute not defined for resource')
+        return self
+
+    def get_permissions(self):
+        delete = self.manager.get_model().delete
+        return ''.join([a.capitalize() for a in delete.split('_')])
+
+    def delete_resources(self, client, delete_args_template, resources):
+        op = getattr(client, self.manager.get_model().delete)
+        delete_args = deepcopy(delete_args_template)
+        for r in resources:
+            delete_args['name'] = r['metadata']['name']
+            if 'namespace' in delete_args_template:
```

Comment:
shouldn't this be looking at the model for namespaced instead of the implicit pass off between subclass and base class as a communication protocol? or put another way is there a particular value for having delete and delete resource as separate classes?

Reply:
one other suggestion on signature, (client, resources, **delete_params), would simplify usage and avoid having to take the dict copy.